sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/policy/RetryPolicy.java

```diff
@@ -249,8 +240,12 @@ private boolean shouldRetryException(RetryStrategy retryStrategy, Exception exce
 
         // Check all causal exceptions in the exception chain.
         while (causalThrowable instanceof IOException || causalThrowable instanceof TimeoutException) {
-            if (retryStrategy.shouldRetryCondition(requestRetryCondition)) {
-                return true;
+            if (shouldRetryCondition != null) {
+                if (shouldRetryCondition.test(requestRetryCondition)) {
+                    return true;
+                }
+            } else {
+                return shouldRetryCondition(requestRetryCondition);
```

Comment:
nit: we should change the name of this method as it gets confusing to see the same name `shouldRetryCondition` used for both the predicate and the private method.